aws_lambda_powertools/utilities/idempotency/exceptions.py

```diff
@@ -3,49 +3,68 @@
 """
 
 
-class IdempotencyItemAlreadyExistsError(Exception):
+class BaseError(Exception):
+    """
+    Base error class that overwrites the way exception and extra information is printed.
+    See https://github.com/awslabs/aws-lambda-powertools-python/issues/1772
+    """
+
+    def format_msg(self, *args: str) -> str:
+        full_msg = str(args[0])
+        if args[1:]:
+            full_msg += f': ({"".join(str(arg) for arg in args[1:])})'
+        return full_msg
+
+    def __init__(self, *args: str):
+        if args:
+            super().__init__(self.format_msg(*args))
+        else:
+            super().__init__()
+
+
```

Comment:
Hello @mploski! This is a huge improvement indeed. I remember the first time I needed to use the idempotency utility, I got an error due to permissions and spent a lot of time figuring out the reason for the "Failed to save record in progress to idempotency store" message. Creating a specific class to capture and return detailed errors makes life easier!!

I was wondering if we really need to instantiate the parent Exception class or if we could change this code to directly return the error message and make the code more clean. But it's just an idea to think about, I'm not sure and would just like to hear your side.

```python
class BaseError(Exception):
     """
    Base error class that overwrites the way exception and extra information is printed.
    See https://github.com/awslabs/aws-lambda-powertools-python/issues/1772
    """

    def __init__(self, *args: str):
        self.message = args[0] if args else ""
        self.args = args

    def __str__(self):
        """
        Return all arguments formatted or original message
        """
        if self.args:
            return f"{' - '.join(str(arg) for arg in self.args)}"
        return self.message
```

Reply:
Hey @leandrodamascena , thx for the feedback! 
I like your proposal and I refactored my code to follow your suggestion i.e to modify how exception is printed in __str__ dunder method. This approach also has additional benefit that we keep initial attributes ( in args list) in object itself so we can easily see how exception object was initialized during debugging if needed.

As for the super class calling. In theory it is suggested to call it as explained in Python Cookbook book: https://books.google.pl/books?id=f7wGeA71_eUC&pg=PA579&dq=To+illustrate+the+use+of+.args,+consider+this+interactive+session+with+the+built-in+RuntimeError+exception,+and+notice+how+any+number+of+arguments+can+be+used+with+the+raise+statement:&hl=en&sa=X&ved=2ahUKEwinm_Dy0bX8AhWMposKHQNHB3oQ6AF6BAgBEAI#v=onepage&q=To%20illustrate%20the%20use%20of%20.args%2C%20consider%20this%20interactive%20session%20with%20the%20built-in%20RuntimeError%20exception%2C%20and%20notice%20how%20any%20number%20of%20arguments%20can%20be%20used%20with%20the%20raise%20statement%3A&f=false. 
I dove deep more to see what happens when we initialize Exception __init__ method. It seems this behavior changed in somewhere in newer python 3 releases and  args list initialization was moved to __new__ method so all needed stuff is initialized either way even without calling superclass init method ( as oppose to python 2.x) . So I think we can safely omit it. 
here is a  c code  for __new__ method in exception class in python 3: https://github.com/python/cpython/blob/3.7/Objects/exceptions.c#L47
and in Python 3.0: 
https://github.com/python/cpython/blob/v3.0/Objects/exceptions.c#L49

Reply:
Hello @mploski! Thank you very much for your feedback and all the links provided, it was very helpful to understand more deeply how the exception is handled and it arguments.

From my side everything is ok and we are ready to merge!